w3af/plugins/crawl/url_payment_finder.py

```diff
@@ -0,0 +1,219 @@
+                desc = 'A potentially interesting url was found : "%s".'
+                desc = desc % response.get_url()
+
+                i = Info('Potentially interesting url', desc, response.id,
+                         self.get_name())
+                i.set_url(response.get_url())
+                
+                kb.kb.append(self, 'url', i)
+                om.out.information(i.get_desc())
+
+                # Report only once
+                self._seen.add(response.get_url())
+
+    def _mutate_path(self, url):
+        """
+        Mutate the path of the url.
+
+        :param url: A URL to transform.
+        :return: A list of URL's that mutate the original url passed
+                 as parameter.
+
+        >>> from w3af.core.data.parsers.doc.url import URL
+        >>> u = url_payment_finder()
+        >>> url = URL( 'http://www.w3af.com/' )
+        >>> list(u._mutate_path(url))
+        []
+
+        >>> URL('http://www.w3af.com/inc/payment.php') in mutants
+        True
+        >>> URL('http://www.w3af.com//module/pay/valid.aspx') in mutants
+        True
+        >>> URL('http://www.w3af.com/cgi-bin/paymentsuccessful.cgi') in mutants
+        True
+        """
+        url_string = url.url_string
+
+        if url_string.count('/') > 2:
+            # Create the new path
+            url_string = url_string[:url_string.rfind('/')]
+
+            dir_to_append_list = self._dirs
+            file_to_append_list = self._files
+            ext_to_append_list = self._exts
+            
+            for dir_to_append in dir_to_append_list:
+                for file_to_append in file_to_append_list:
+                    for ext_to_append in ext_to_append_list:
+                        newurl = URL(url_string + dir_to_append + file_to_append + ext_to_append)
+                        yield newurl
+
+    def _verify_head_enabled(self, url):
+        """
+        Verifies if the requested URL permits a HEAD request.
+        This was saved inside the KB by the plugin allowed_methods
+
+        :return : Sets self._head to the correct value, nothing is returned.
+        """
+        allowed_methods_infos = kb.kb.get('allowed_methods', 'methods')
+        allowed_methods = []
+        for info in allowed_methods_infos:
+            allowed_methods.extend(info['methods'])
+        
+        if 'HEAD' in allowed_methods:
+            self._head = True
+        else:
+            self._head = False
+
+    def _head_enabled(self):
+        return self._head
+
+    def get_plugin_deps(self):
+        """
+        :return: A list with the names of the plugins that should be run before the
+        current one.
+        """
+        return ['infrastructure.allowed_methods']
+
+    def get_long_desc(self):
+        """
+        :return: A DETAILED description of the plugin functions and features.
+        """
+        return """
+        This plugin will try to find new URL's based on the input. If the input
```

Comment:
`This plugin will try to find new payment gateway webhook URLs based on the input`

Reply:
Done